core/src/main/scala/com/salesforce/op/evaluators/OpRegressionEvaluator.scala

```diff
@@ -55,6 +60,50 @@ private[op] class OpRegressionEvaluator
 
   @transient private lazy val log = LoggerFactory.getLogger(this.getClass)
 
+  final val signedPercentageErrorHistogramBins = new DoubleArrayParam(
+    parent = this,
+    name = "signedPercentageErrorHistogramBins",
+    doc = "sequence of bins for signed percentage error histogram",
+    isValid = l => l.nonEmpty && (l sameElements l.sorted)
+  )
+  setDefault(signedPercentageErrorHistogramBins,
+    Array(Double.NegativeInfinity) ++ (-100.0 to 100.0 by 10) ++ Array(Double.PositiveInfinity)
+  )
+
+  def setPercentageErrorHistogramBins(v: Array[Double]): this.type = set(signedPercentageErrorHistogramBins, v)
+
+  final val scaledErrorCutoff = new DoubleParam(
+    parent = this,
+    name = "scaledErrorCutoff",
+    doc = "label value cutoff below which percentage error is implemented as a scaled error " +
+      "with a fixed denominator to avoid problems with label values around 0",
+    isValid = (d: Double) => d > 0.0
+  )
+  setDefault(scaledErrorCutoff, 1E-3)
+
+  def setScaledErrorCutoff(v: Double): this.type = set(scaledErrorCutoff, v)
+  def getScaledErrorCutoff: Option[Double] = get(scaledErrorCutoff)
+
+  final val smartCutoff = new BooleanParam(
```

Comment:
not sure you need this boolean param it seems overkill/harder to keep track of everything. perhaps just dont have a default for the smart cutoff value and if it is set use that instead of the scaled error cutoff.

Reply:
Yes, I was toying with this idea already. Let me implement it. 👍

Reply:
Done. It was easier than expected.